File: OwnHabits.Domain/Models/Characteristic.cs
using System.Collections.Generic;
using OwnHabits.Domain.Common;
using OwnHabits.Domain.Models.User;

namespace OwnHabits.Domain.Models;

public class Characteristic : BaseTextEntity
{
    public int Level { get; set; } = 1;
    public int CurrentExperience { get; set; } = 0;
    public int NextLevelExperience { get; set; } = 100;

    public virtual ICollection<SkillToCharacteristics> RequiredSkills { get; set; } = [];
    public virtual ICollection<Achievement> CompletedAchievements { get; set; } = [];
    public virtual ICollection<UserToCharacteristics> UserCharacteristics { get; set; } = [];

}

File: Frontend/own-habits-frontend/src/types/Characteristic.ts
import type { Achievement } from './Achievement.tsx';
import type { SkillToCharacteristics } from './SkillToCharacteristics.ts';
import type { UserToCharacteristics } from './UserToCharacteristics.ts';

export interface Characteristic {
  id: string;
  userId: string;
  title: string;
  description?: string;
  level: number;
  currentExperience: number;
  nextLevelExperience: number;
  
  // Навигационные свойства
  requiredSkills: SkillToCharacteristics[];
  completedAchievements: Achievement[];
  userCharacteristics: UserToCharacteristics[];
}